### PR TITLE
🐛 FIX: cli.parse: return 0 for non-error #53

### PR DESCRIPTION
**Pull request by westurner**
This is necessary so that make doesn't fail to execute this task due to `markdown-it` returning True (1) instead of zero (0) for success:

```makefile
header-footer-templates:
	@#pip install markdown-it-py
	markdown-it README.md.header.md > index.html.header.html
	markdown-it README.md.footer.md > index.html.footer.html

```

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -2,14 +2,21 @@
...
     with tempfile.TemporaryDirectory() as tempdir:
         path = pathlib.Path(tempdir).joinpath("test.md")
         path.write_text("a b c")
-        assert parse.main([str(path)])
+        assert parse.main([str(path)]) == 0
+
+
+def test_parse_fail():
+    with pytest.raises(SystemExit):
+        assert parse.main(["/tmp/nonexistant_path/for_cli_test.md"]) == 1
```
Comment (chrisjsewell) [inline tests/test_cli.py:19]: nitpick, but surely you will never actually test this assert, because the parse raises a `SystemExit`

Comment (westurner) [reply tests/test_cli.py:19]: Yeah, but it's helpful to document that the returncode when SystemExit is raised is 1.

Is there branch coverage on the tests/?

Comment (westurner) [reply tests/test_cli.py]: I've removed the assert.

https://docs.python.org/3/library/sys.html#sys.exit
https://docs.python.org/3/library/exceptions.html#SystemExit

File: tests/test_cli.py [hunk 2]
```diff
@@ -2,14 +2,21 @@
...
     with tempfile.TemporaryDirectory() as tempdir:
         path = pathlib.Path(tempdir).joinpath("test.md")
         path.write_text("a b c")
-        assert parse.main([str(path)])
+        assert parse.main([str(path)]) == 0
+
+
+def test_parse_fail():
+    with pytest.raises(SystemExit):
+        parse.main(["/tmp/nonexistant_path/for_cli_test.md"])
```
Comment (hukkinj1) [inline tests/test_cli.py:19]: SystemExit per se doesn't mean that exit code is non-zero. We can assert that with something like:
```suggestion
    with pytest.raises(SystemExit) as exc_info:
        parse.main(["/tmp/nonexistant_path/for_cli_test.md"])
    assert exc_info.value.code == 1
```

File: tests/test_cli.py [hunk 3]
```diff
@@ -2,14 +2,22 @@
...
         path = pathlib.Path(tempdir).joinpath("test.md")
         path.write_text("a b c")
-        assert parse.main([str(path)])
+        assert parse.main([str(path)]) == 0
+
+
+def test_parse_fail():
+    with pytest.raises(SystemExit) as exc_info:
+        parse.main(["/tmp/nonexistant_path/for_cli_test.md"])
+    assert exc_info.value.code == 1
```
Comment (hukkinj1) [inline tests/test_cli.py:20]: One way to fix the test failure is:
```suggestion
    assert isinstance(exc_info.value.code, str)
```
---
Another way, that I personally would prefer but don't know if @chrisjsewell would, is go to `markdown_it.cli.parse.convert_file` and replace
```python
sys.exit('Cannot open file "{}".'.format(filename))
```
with
```python
sys.stderr.write('Cannot open file "{}".\n'.format(filename))
sys.exit(1)
```

**Comment by chrisjsewell** [comment]
Superseded by #124

**Comment by westurner** [comment]
Thanks

**Pull request closed without merging**